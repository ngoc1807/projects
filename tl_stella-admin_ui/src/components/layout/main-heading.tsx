import { FC } from "react";
import { Typography, Divider, Box, SxProps, Theme } from "@mui/material";
import { CustomSkeleton } from "src/components/custom-skeleton/custom-skeleton";

interface Props {
  disableUnderline?: boolean;
  children: string | undefined;
  sx?: SxProps<Theme>;
}

export const MainHeading: FC<Props> = ({
  disableUnderline = false,
  children,
  sx,
}) => {
  // SKELETON: uncomment this to show skeleton load
  // children = undefined;

  return (
    <Box sx={sx}>
      {children !== undefined ? (
        <Typography
          variant="h5"
          component="h1"
          sx={{
            display: "inline",
          }}
        >
          {children}
        </Typography>
      ) : (
        <CustomSkeleton width={400} height={43} />
      )}
      {disableUnderline === false && <Divider sx={{ mt: 2, mb: 5 }} />}
    </Box>
  );
};
