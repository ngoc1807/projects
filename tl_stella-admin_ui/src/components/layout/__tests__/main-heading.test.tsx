import { cleanup, render } from "@testing-library/react";
import { MainHeading } from "../main-heading";

describe("main-heading", () => {
  afterEach(() => {
    cleanup();
  });
  it("should render correctly", () => {
    const { container } = render(<MainHeading>test</MainHeading>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
