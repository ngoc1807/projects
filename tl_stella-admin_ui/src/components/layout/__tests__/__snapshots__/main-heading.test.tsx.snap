// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main-heading should render correctly 1`] = `
<div
  class="MuiBox-root css-0"
>
  <h1
    class="MuiTypography-root MuiTypography-h5 css-1lyr4fm-MuiTypography-root"
  >
    test
  </h1>
  <hr
    class="MuiDivider-root MuiDivider-fullWidth css-1e2tz01-MuiDivider-root"
  />
</div>
`;
